src/Files/File.php

```diff
@@ -2355,109 +2355,153 @@ public function getCondition($stackPtr, $type)
 
     /**
      * Returns the name of the class that the specified class extends.
-     * (works for classes, anonymous classes and interfaces)
+     *
+     * Works for classes, anonymous classes and interfaces, though it is
+     * strongly recommended to use the findExtendedInterfaceNames() method
+     * to examine interfaces as they can extend multiple parent interfaces.
      *
      * Returns FALSE on error or if there is no extended class name.
      *
-     * @param int $stackPtr The stack position of the class.
+     * @param int $stackPtr The stack position of the class keyword.
      *
      * @return string|false
      */
     public function findExtendedClassName($stackPtr)
     {
-        // Check for the existence of the token.
-        if (isset($this->tokens[$stackPtr]) === false) {
-            return false;
-        }
+        $validStructures = [
+            T_CLASS      => true,
+            T_ANON_CLASS => true,
+            T_INTERFACE  => true,
+        ];
 
-        if ($this->tokens[$stackPtr]['code'] !== T_CLASS
-            && $this->tokens[$stackPtr]['code'] !== T_ANON_CLASS
-            && $this->tokens[$stackPtr]['code'] !== T_INTERFACE
-        ) {
-            return false;
-        }
+        $classes = $this->examineObjectDeclarationSignature($stackPtr, $validStructures, T_EXTENDS);
 
-        if (isset($this->tokens[$stackPtr]['scope_opener']) === false) {
+        if (empty($classes) === true) {
             return false;
+        } else {
+            // Classes can only extend one parent class.
+            return $classes[0];
         }
 
-        $classOpenerIndex = $this->tokens[$stackPtr]['scope_opener'];
-        $extendsIndex     = $this->findNext(T_EXTENDS, $stackPtr, $classOpenerIndex);
-        if (false === $extendsIndex) {
-            return false;
-        }
+    }//end findExtendedClassName()
 
-        $find = [
-            T_NS_SEPARATOR,
-            T_STRING,
-            T_WHITESPACE,
-        ];
 
-        $end  = $this->findNext($find, ($extendsIndex + 1), ($classOpenerIndex + 1), true);
-        $name = $this->getTokensAsString(($extendsIndex + 1), ($end - $extendsIndex - 1));
-        $name = trim($name);
+    /**
+     * Returns the names of the interfaces that the specified interface extends.
+     *
+     * Returns FALSE on error or if there is no extended interface name.
+     *
+     * @param int $stackPtr The stack position of the interface keyword.
+     *
+     * @return array|false
+     */
+    public function findExtendedInterfaceNames($stackPtr)
+    {
+        $validStructures = [T_INTERFACE => true];
 
-        if ($name === '') {
+        $interfaces = $this->examineObjectDeclarationSignature($stackPtr, $validStructures, T_EXTENDS);
+
+        if (empty($interfaces) === true) {
```

Comment:
From the end-user (= sniff developers) point of view, the difference is negligible, `if (empty($interfacesNames))` will catch both `false` as well as en empty array.

What's better is a matter of opinion and in this case, I've elected consistency with the similar methods over always having the same type of return.

It can be adjusted easily enough and I'd be happy to do so if @gsherwood would deem always returning an array appropriate.